### PR TITLE
Charlesmchen/improve attachment approval view

### PR DESCRIPTION
**Pull request by charlesmchen**
*	Add video preview to attachment approval dialog. 
*	Add animated image preview to attachment approval dialog.
*	Add stubs for audio preview to attachment approval dialog. 
*	Improve file size formatting in attachment approval dialog.  

PTAL @michaelkirk 

screen shot 2017-04-12 at 4 52 57 pm
screen shot 2017-04-12 at 4 36 49 pm
screen shot 2017-04-12 at 4 36 33 pm

### CODE REVIEW

File: Signal/src/ViewControllers/AttachmentApprovalViewController.swift [hunk 1]
```diff
@@ -60,29 +63,75 @@ class AttachmentApprovalViewController: UIViewController {
...
-        if image != nil {
-            let imageView = UIImageView(image:image)
-            imageView.layer.minificationFilter = kCAFilterTrilinear
-            imageView.layer.magnificationFilter = kCAFilterTrilinear
-            imageView.contentMode = .scaleAspectFit
-            attachmentPreviewView.addSubview(imageView)
-            imageView.autoPinWidthToSuperview()
-            imageView.autoPinHeightToSuperview()
-        } else {
+        guard image != nil else {
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/AttachmentApprovalViewController.swift]: ooo so swifty 😎😎😎

File: Signal/src/ViewControllers/AttachmentApprovalViewController.swift [hunk 2]
```diff
@@ -125,9 +174,17 @@ class AttachmentApprovalViewController: UIViewController {
...
+        let fileSize = attachment.data.count
+        let kOneKilobyte = 1024
+        let kOneMegabyte = kOneKilobyte * kOneKilobyte
+        let fileSizeText = (fileSize > kOneMegabyte
+            ? numberFormatter.string(from: NSNumber(value: fileSize / kOneMegabyte))! + " mb"
+            : (fileSize > kOneKilobyte
+                ? numberFormatter.string(from: NSNumber(value: fileSize / kOneKilobyte))! + " kb"
+                : numberFormatter.string(from: NSNumber(value: fileSize))!))
         fileSizeLabel.text = String(format:NSLocalizedString("ATTACHMENT_APPROVAL_FILE_SIZE_FORMAT",
                                                              comment: "Format string for file size label in call interstitial view"),
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/AttachmentApprovalViewController.swift]: nit: could you update the comment to tell translators what they're formatting, e.g. "embeds {{file size as "x mb" or "x kb" }}'

File: Signal/src/ViewControllers/SignalAttachment.swift [hunk 3]
```diff
@@ -109,6 +111,29 @@ class SignalAttachment: NSObject {
...
+                return nil
+            }
+            let fileName = NSUUID().uuidString + "." + fileExtension
+            guard let fileUrl = NSURL.fileURL(withPathComponents: [directory, fileName]) else {
+                return nil
+            }
+            do {
+                try data.write(to: fileUrl)
+            } catch {
+                Logger.info("\(SignalAttachment.TAG) Could not write data to disk: \(dataUTI)")
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/SignalAttachment.swift]: nit: can you add an Logger.error/assert

File: Signal/src/ViewControllers/AttachmentApprovalViewController.swift (hunk 2 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/AttachmentApprovalViewController.swift]: Done.

File: Signal/src/ViewControllers/SignalAttachment.swift (hunk 3 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/SignalAttachment.swift]: Done.

**Pull request merged**